### DISCUSSION TITLE
Webpack html-inline-script-webpack-plugin do not work after upgade pixi v.8

### DISCUSSION THREAD

**Discussion by jkukuryk** [pixijs/pixijs#10344] [v8 General]
When I updated the Pixi version, the webpack plugin html-inline-script-webpack-plugin created 4/5 additional chunks.
- all assets and other elements are packed in HTML
- only pixijs files go to additional chunks.

Plugin works perfectly with pixi v. 7

GOAL: I must build a single HTML file with all the code and assets.

Plugin: https://github.com/icelam/html-inline-script-webpack-plugin

**Comment by jkukuryk** [comment]
This is the repo with a basic webpack and pixi configuration:
https://github.com/jkukuryk/pixi-version-update

**Reply by bigtimebuddy** [reply]
In PixiJS v8, we are using a lot more dynamic imports to help minimize the amount of code that needs to be loaded up front. In order to inline all the code, you need a way to disable dynamic imports. Perhaps a feature request for the Webpack plugin?

**Reply by jkukuryk** [reply]
Hi Matt, Thank you for Your reply. The idea of minimizing code requests is great. Unfortunately, I need to create an HTML ad with a single index.html.  I'll try to ask on this webpack plugin.

**Reply by jkukuryk** [reply]
I created a related topic on the webpack inline plugin:
https://github.com/icelam/html-inline-script-webpack-plugin/issues/494

**Comment by phloud** [comment]
hi Jakub, I'm in the same boat making playables, been using v7 for ages and had a poke at v8 and ran into this issue. wondered if you ever found a solution?

**Reply by jkukuryk** [reply]
@phloud Hi, I created a technical task for this week to return to this topic. I started experimenting with Vite for my BabylonJS projects, so I'll try to test those solutions on Pixi. I'll put updates here

**Reply by jkukuryk** [reply]
@phloud Hi it has been a while.
I created my first playable with Pixi8, but I had to switch to Vite and abandon webpack.

**Comment by jkukuryk** [comment]
My solution to single file with pixi8 and Vite:
 - use all css inside js with `import './index.css';`
 
 My project template structure is build with subrepo witch is attached inside engine folder with some build templates html.
 But I hope you would be able to figure out your paths:
 ```

import { defineConfig } from 'vite';
import { createHtmlPlugin } from 'vite-plugin-html';
import { viteSingleFile } from 'vite-plugin-singlefile';
import path from 'path';
import { getProdFileName, renameOutputFiles } from './scripts/fileName.mjs'; 

export default defineConfig({
    base: '',  // important 
    assetsInclude: ['**/*.glb'], // I use pixi with three.js but here you can add more extensions if needed

    build: {
        target: 'es2015',
        outDir: 'dist',
        emptyOutDir: false,
        assetsInlineLimit: Infinity,  // => crusial for inline
        rollupOptions: {
            treeshake: true,
            input: path.resolve(__dirname, '../../src/index.ts'),  // your main entry file 
            output: {
                inlineDynamicImports: true,  // inline imports
                manualChunks: undefined,
                entryFileNames: 'index.js',  //not important for single file
            },
        },
        minify: 'terser',
        terserOptions: {
            compress: {
                drop_console: true,
                drop_debugger: true,
                passes: 3,
            },
            format: {
                comments: false,
            },
        },
    },
    resolve: {
        alias: {
            assets: path.resolve(__dirname, '../../assets'),  // aliases works like in webpack
            engine: path.resolve(__dirname, '../'),
            src: path.resolve(__dirname, '../../src'),
            root: path.resolve(__dirname, '../../'),
        },
        extensions: ['.ts', '.js', '.json'],
    },
    plugins: [
        createHtmlPlugin({
            template: 'engine/build/public/unity.html',   // here inject html templates
        }),
        viteSingleFile(),   // run plugin
        renameOutputFiles(getProdFileName('Unity'), 'unity'),  // no matter what would the files would be named as an HTML template, so there is a need for an additional plugin to rename files, if you don't want to do it manually.
        
    ],
});
```
